src/Microsoft.AspNet.Mvc.Core/Routing/AttributeRouteMatchingEntry.cs

```diff
@@ -21,9 +24,20 @@ public class AttributeRouteMatchingEntry
         /// </summary>
         public decimal Precedence { get; set; }
 
-        /// <summary>
-        /// The <see cref="TemplateRoute"/>.
-        /// </summary>
-        public TemplateRoute Route { get; set; }
+        public IRouter Target { get; set; }
+
+        public string RouteName { get; set; }
+
+        public string RouteTemplate { get; set; }
+
+        public TemplateMatcher TemplateMatcher { get; set; }
+
+        public IReadOnlyDictionary<string, object> Defaults { get; set; }
+
+        public IReadOnlyDictionary<string, IRouteConstraint> Constraints { get; set; }
+
+        public IReadOnlyDictionary<string, object> DataTokens { get; set; }
+
+        public IInlineConstraintResolver InlineConstraintResolver { get; set; }
```

Comment:
This shouldn't be needed, since `Constraints` already has them resolved. By wary of passing around services like this, it usually means something needs a refactor.